### ISSUE TITLE
Hessian Vector Product with vmap?

### ISSUE THREAD

**Issue by pp500** [google/jax#2937]
I'd like to compute a hessian vector product on several points and i'd like to avoid using a for loop.
I found how to compute the gradient and the hessian (example below) but i can't get the hessian vector product to work.

I attach a small example below

Any ideas? 

```python
import jax.numpy as np
from jax import grad, vmap, hessian
from jax import random 

def example(X):
    f=X[0]**2+X[1]**2
    return f

key = random.PRNGKey(0)
N=5
d=2
X_t=random.normal(key, (N,d))
W_t=random.normal(key, (N,d))
V=example
dV=vmap(grad(V))
ddV=vmap(hessian(V))
dV_t=dV(X_t)
```

**Comment by shoyer** [comment]
What's specifically is wrong with your code example?

Have you tried the `hvp` example from the Autodiff cookbook?
https://jax.readthedocs.io/en/latest/notebooks/autodiff_cookbook.html#Hessian-vector-products-with-grad-of-grad

**Comment by mattjj** [comment]
Actually this later section in the autodiff cookbook is better:

https://jax.readthedocs.io/en/latest/notebooks/autodiff_cookbook.html#Hessian-vector-products-using-both-forward--and-reverse-mode

```python
def hvp(f, primals, tangents):
  return jvp(grad(f), primals, tangents)[1]
```

**Comment by pp500** [comment]
Great thanks! 
To clarify, here is my question with the hvp definition above

```python
import jax.numpy as np
from jax import grad,jvp,vmap
from jax import random 

def example(X):
    f=X[0]**2+X[1]**2
    return f

def hvp(f, primals, tangents):
  return jvp(grad(f), primals, tangents)[1]

key = random.PRNGKey(0)
N=5
d=2

X_t=random.normal(key, (N,d))
W_t=random.normal(key, (N,d))

V=example
dV=vmap(grad(V))
ddV=vmap(hessian(V))

dV_t=dV(X_t)
ddV_t=ddV(X_t)

ans1 = hvp(V, (X_t[0],), (W_t[0],))
```

dv_t (gradient) and ddV_t (Hessian) are vectorised evaluations.
But for the hessian vector product I can't find a way to vectorise it.
ans1 above contains the correct hessian vector product but it is only the first one.
Is it possible to vectorise ans1 above?

**Comment by shoyer** [comment]
Maybe something like:
```python
from functools import partial

@partial(vmap, in_axes=(0, 0))
def vectorized_hvp(X_t, W_t):
  return hvp(V, (X_t,), (W_t,))

ans2 = vectorized_hvp(X_t, W_t)
```

**Comment by pp500** [comment]
Great, thank you! That does the job!
